engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/batch/NadelBatchHydrationByObjectId.kt

```diff
@@ -0,0 +1,144 @@
+package graphql.nadel.enginekt.transform.hydration.batch
+
+import graphql.nadel.ServiceExecutionResult
+import graphql.nadel.enginekt.blueprint.NadelBatchHydrationFieldInstruction
+import graphql.nadel.enginekt.blueprint.NadelOverallExecutionBlueprint
+import graphql.nadel.enginekt.blueprint.hydration.NadelBatchHydrationMatchStrategy
+import graphql.nadel.enginekt.blueprint.hydration.NadelHydrationActorInputDef
+import graphql.nadel.enginekt.transform.NadelTransformUtil
+import graphql.nadel.enginekt.transform.hydration.NadelHydrationUtil.getHydrationActorNodes
+import graphql.nadel.enginekt.transform.query.NadelQueryPath
+import graphql.nadel.enginekt.transform.result.NadelResultInstruction
+import graphql.nadel.enginekt.transform.result.asMutable
+import graphql.nadel.enginekt.transform.result.json.JsonNode
+import graphql.nadel.enginekt.transform.result.json.JsonNodeExtractor
+import graphql.nadel.enginekt.util.AnyList
+import graphql.nadel.enginekt.util.AnyMap
+import graphql.nadel.enginekt.util.JsonMap
+import graphql.nadel.enginekt.util.asNullableJsonMap
+import graphql.nadel.enginekt.util.emptyOrSingle
+import graphql.nadel.enginekt.util.flatten
+import graphql.nadel.enginekt.util.isList
+import graphql.nadel.enginekt.util.queryPath
+import graphql.nadel.enginekt.util.unwrapNonNull
+
+internal object NadelBatchHydrationByObjectId {
```

Comment:
Nothing changed here, this code is stripped from `NadelBatchHydrator` exactly as it was.